pkg/controllers/user/logging/projectlogging.go

```diff
@@ -50,33 +51,42 @@ func registerProjectLogging(cluster *config.UserContext) {
 		clusterLister:        cluster.Management.Management.Clusters("").Controller().Lister(),
 		clusterName:          cluster.ClusterName,
 	}
-	projectLoggings.AddClusterScopedHandler("project-logging-controller", cluster.ClusterName, syncer.Sync)
+	projectLoggings.AddClusterScopedLifecycle("project-logging-controller", cluster.ClusterName, lifecycle)
 }
 
-func (c *ProjectLoggingSyncer) Sync(key string, obj *v3.ProjectLogging) error {
-	//clean up
-	if obj == nil || obj.DeletionTimestamp != nil {
-		return utils.CleanResource(c.namespaces, c.clusterLoggingLister, c.projectLoggings.Controller().Lister())
-	}
+func (c *ProjectLoggingLifecycle) Create(obj *v3.ProjectLogging) (*v3.ProjectLogging, error) {
+	newObj, err := v3.LoggingConditionProvisioned.DoUntilTrue(obj, func() (runtime.Object, error) {
+		return obj, c.provision(obj)
+	})
+	return newObj.(*v3.ProjectLogging), err
+}
+func (c *ProjectLoggingLifecycle) Updated(obj *v3.ProjectLogging) (*v3.ProjectLogging, error) {
+	newObj, err := v3.LoggingConditionProvisioned.DoUntilTrue(obj, func() (runtime.Object, error) {
+		return obj, c.provision(obj)
+	})
 
-	if err := utils.IniteNamespace(c.namespaces); err != nil {
-		return err
-	}
-	if err := utils.InitConfigMap(c.configmaps); err != nil {
-		return err
-	}
-	if err := c.createOrUpdateProjectConfigMap(); err != nil {
-		return err
+	if err != nil {
+		return newObj.(*v3.ProjectLogging), err
 	}
 
-	if err := utils.CreateLogAggregator(c.daemonsets, c.serviceAccounts, c.clusterRoleBindings, c.clusterLister, c.clusterName, loggingconfig.LoggingNamespace); err != nil {
-		return err
-	}
+	newObj, err = v3.LoggingConditionUpdated.Do(newObj, func() (runtime.Object, error) {
+		if err := c.createOrUpdateProjectConfigMap(); err != nil {
+			return obj, err
+		}
+		return obj, nil
+	})
 
-	return utils.CreateFluentd(c.daemonsets, c.serviceAccounts, c.clusterRoleBindings, loggingconfig.LoggingNamespace)
+	return newObj.(*v3.ProjectLogging), err
+}
+
+func (c *ProjectLoggingLifecycle) Remove(obj *v3.ProjectLogging) (*v3.ProjectLogging, error) {
+	if err := utils.UnsetConfigMap(c.configmaps, loggingconfig.ProjectLoggingName, "project"); err != nil {
```

Comment:
If there is multiple project, delete one project should not set the `project.conf` to empty.